tests/codegen/naked-fn/aligned.rs

```diff
@@ -0,0 +1,20 @@
+//@ compile-flags: -C no-prepopulate-passes -Copt-level=0
+//@ needs-asm-support
+//@ only-x86_64
+
+#![crate_type = "lib"]
+#![feature(naked_functions, fn_align)]
+use std::arch::asm;
+
+// CHECK: Function Attrs: naked
+// CHECK-NEXT: define{{.*}}void @naked_empty()
+// CHECK: align 16
+#[repr(align(16))]
```

Comment:
What about other reprs?

Reply:
Fair enough. Ill add a test but theres no reason it shouldn't work. Gimme till after dinner haha

Reply:
`fn_align` should only bless `repr(align(N))`, it shouldn't bless e.g. `repr(C)`.

Reply:
Right, so then it (that attr) will already make sure it's correct

Reply:
sure, but I can write `#[repr(C)]` on stable.

Reply:
And this PR allows that, presumably now, so it should be tested

Reply:
er, actually I got my fn reprs and my struct reprs mixed up

Reply:
Nothing needs modified here anymore.

Reply:
No worries haha, leave it as is then?